packages/jest-haste-map/src/index.js

```diff
@@ -325,6 +325,28 @@ class HasteMap extends EventEmitter {
           throw new Error(message);
         }
         this._console.warn(message);
+        // We do NOT want consumers to use a module that is ambiguous.
+        delete moduleMap[platform];
+        if (Object.keys(moduleMap).length === 1) {
+          delete map[id];
+        }
+        let dupsByPl = hasteMap.duplicates[id];
+        if (dupsByPl == null) {
+          dupsByPl = hasteMap.duplicates[id] = (Object.create(null): any);
+        }
```

Comment:
Ah, now I get it. This is because at that point the original module won't be part of the moduleMap, so the previous if-statement with `existingModule && existingModule[H.PATH] !== module[H.PATH]` will be false.